### ISSUE TITLE
Test swallows stdout/stderr

### ISSUE THREAD

**Issue by bbarker** [joelburget/easytest#25]
At least, it seems to for me. This is problematic when attempting print-style deubgging.

**Comment by bbarker** [comment]
Saw this, which I thought might be related: https://github.com/hedgehogqa/haskell-hedgehog/issues/236 But it looks like SIlently/capture is not being used in easytest

**Comment by bbarker** [comment]
I also tried running e.g. `hFlush stderr` just to be sure, but did not seem to help

**Comment by bbarker** [comment]
Correction: `hFlush` does help, but only for `stdout`, not `stderr`.

**Comment by bbarker** [comment]
(which seems to create problems when trying to use Debug.Trace's `trace`, as it relies on [debugBelch](https://github.com/ghc/ghc/blob/c5911479f295242e16e396eb5d1369f2e4ce8de0/includes/rts/Messages.h#L86), which seems to have hardcoded STDERR.)

**Comment by joelburget** [comment]
Right, this is related to that hedgehog issue in that hedgehog overwrites portions of the console with output. If you're using `print` / `putStr` in `IO`, using silently should fix your problem. I haven't yet figured out how to use `trace` without it being overwritten, though I hope the response to my question in the thread you linked will help.

**Comment by bbarker** [comment]
@joelburget thanks for pointing it out - wasn't quite sure how it was related at first glance, but I tried it out ... but now I'm even more confused, though tentatively in a pleasant way, because it actually *fixed* my failing test! more investigation on the way to see if it can fix my actual code. But unfortunately, I still didn't see the msg when used like : `evaluate $ trace msg $ action`.

**Comment by bbarker** [comment]
OK, it didn't actually fix the test either - it just caused it to not surface the exception, which is sort of the opposite of the intent of `evaluate`; if I do `join $ evaluate $ ... ` I'm back to my original error - I think this is because my `...` action was returning an `IO` itself.
